### PR TITLE
Fix Cartesian Product, Support Ensemble ID Queries, and Update Validator, Preprocess schema

### PR DESCRIPTION
**Pull request by DeazStar**
* Fixed cartesian product in cypher queries 
* Handled Ensemble ID queries
* Update validator for handling request without predicates
* Processed schema even further
* update metta query generator based on the new schema

### CODE REVIEW

File: app/lib/validator.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ def validate_request(request, schema):
                 raise Exception(f"{property} doesn't exsist in the schema!")
```
Comment (dawit-melka) [inline app/lib/validator.py:29]: For the time being please comment out validation for properties in node. There needs to be some updates to be done in the schema

Comment (dawit-melka) [reply app/lib/validator.py:29]: also if 'properties' field is not present handling it by adding empty object would be better
```
if 'properties' not in node:
            raise Exception("properties is required")
```

File: app/services/cypher_generator.py [hunk 2]
```diff
@@ -64,65 +64,85 @@ def run_query(self, query_code):
...
+            for node in nodes:
+                var_name = f"n_{node['node_id']}"
+                match_no_preds.append(self.match_node(node, var_name))
+                return_no_preds.append(var_name)
+            cypher_query = self.construct_clause(match_no_preds, return_no_preds)
             cypher_queries.append(cypher_query)
         else:
-            logging.debug("Processing stopped due to invalid request.")
+            for i, predicate in enumerate(predicates):
+                predicate_type = predicate['type'].replace(" ", "_")
```
Comment (dawit-melka) [inline app/services/cypher_generator.py:95]: To avoid potential case-sensitivity issues, it's recommended to apply `.lower()` to the predicate.

**Review by dawit-melka — CHANGES_REQUESTED** [review]

File: app/lib/validator.py (hunk 1 above)
Comment (DeazStar) [reply app/lib/validator.py:30]: done

File: app/services/cypher_generator.py (hunk 2 above)
Comment (DeazStar) [reply app/services/cypher_generator.py]: done

**Comment by dawit-melka** [comment]
Looks good!

**Pull request merged**